scripts/pages/zoning/zoning.ts

```diff
@@ -728,110 +725,17 @@ class ZoneMenuHandler {
 		}
 	}
 
-	onPointPicked(region: Region) {
-		if (!this.selectedZone || !this.selectedZone.region) return;
-
-		switch (this.pointPick) {
-			case PickType.NONE:
-				this.showInfoPanel(false);
-				return;
-
-			case PickType.CORNER:
-				this.selectedZone.region.points = region.points;
-				this.selectedZone.region.bottom = region.bottom;
-				this.panels.regionBottom.text = this.selectedZone.region.bottom.toFixed(2);
-				break;
-
-			case PickType.BOTTOM:
-				this.selectedZone.region.bottom = region.bottom;
-				this.panels.regionBottom.text = this.selectedZone.region.bottom.toFixed(2);
-				break;
-
-			case PickType.HEIGHT:
-				this.selectedZone.region.height = region.height;
-				this.panels.regionHeight.text = this.selectedZone.region.height.toFixed(2);
-				break;
-
-			case PickType.SAFE_HEIGHT:
-				this.selectedZone.region.safeHeight = region.safeHeight;
-				this.panels.regionSafeHeight.text = this.selectedZone.region.safeHeight.toFixed(2);
-				break;
-
-			case PickType.TELE_DEST_POS:
-				this.selectedZone.region.teleDestPos = region.teleDestPos;
-				this.panels.regionTPPos.x.text = this.selectedZone.region.teleDestPos[0].toFixed(2);
-				this.panels.regionTPPos.y.text = this.selectedZone.region.teleDestPos[1].toFixed(2);
-				this.panels.regionTPPos.z.text = this.selectedZone.region.teleDestPos[2].toFixed(2);
-				break;
-
-			case PickType.TELE_DEST_YAW:
-				this.selectedZone.region.teleDestYaw = region.teleDestYaw;
-				this.panels.regionTPYaw.text = this.selectedZone.region.teleDestYaw.toFixed(0);
-				break;
-		}
-
-		this.drawZones();
-	}
-
-	onPointDeleted(newPoints: Region) {
-		if (!this.selectedZone || !this.selectedZone.region) return;
-
-		this.selectedZone.region = newPoints;
+	onRegionEditCompleted(newRegion: Region) {
+		const index = this.panels.regionSelect.GetSelected()?.GetAttributeInt('value', -1);
+		if (index !== -1) this.selectedZone.zone.regions[index] = newRegion;
 
+		this.showInfoPanel(false);
 		this.drawZones();
 	}
 
-	onPickCanceled() {
-		if (!this.isSelectionValid().zone || !this.selectedZone.region) return;
+	onRegionEditCanceled() {
+		if (!this.selectedZone?.region?.points.length) this.deleteRegion();
```

Comment:
I think points is missing a ?

Reply:
similar to the other comment, a `region` should always have a `points` property even if it's empty. Is there a better way to have an assert type error-popup behavior?

Reply:
Panzer is right, as above, ?. doesn't evaluate to like, some default initialized struct, it just creates an `undefined`. There's practically never a case where you should do `A?.B.C`, in fact TS strict mode will flag any instance of that, assuming `A` is actually possibly `undefined` or `null`.